src/js/common/utils/accessible-menus.js

```diff
@@ -173,9 +173,9 @@ function closeMenu(menuLi) {
  *                                 second-level menu in a wide screen, we want the third-level opened as well
  * @param {bool} stealFocus       Whether or not to focus on the first item in the opened menu
  */
-function openMenu(menuLi, openSubMenu = false, stealFocus = true) {
+function openMenu(menuLi, menuStructure = null, openSubMenu = false, stealFocus = true) {
   // If we're not dealing with a menu structure, abort
-  const struct = getMenuStructure(menuLi);
+  const struct = menuStructure || getMenuStructure(menuLi);
```

Comment:
Is it of interest to make the logic and and on line 155 the same, they appear to do the same thing differently, or even to pull this logic out into a helper and use in both places?

Reply:
Yeah, I oscillated on this one, but decided to leave `closeMenu` (hopefully) simpler with just one param. I gave `openMenu` both `menuLi` and `menuStructure = null` because we actually use `menuLi` for more than just getting the structure of the menu down on line 186. In `closeMenu`, it just needs the structure, not the `menuLi`.

Hopefully that made sense, but if not, we I can try to clarify some more.

I'm having a hard time with words this morning. 😞